src/PgRvn.Server/PowerBI/PBIAllCollectionsQuery.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Raven.Client.Documents;
+using Raven.Client.Documents.Operations;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PgRvn.Server
+{
+    public class PBIAllCollectionsQuery : RqlQuery
+    {
+        public PBIAllCollectionsQuery(string queryText, int[] parametersDataTypes, IDocumentStore documentStore) 
+            : base(queryText, parametersDataTypes, documentStore)
+        {
+        }
+
+        public static bool TryParse(string queryText, int[] parametersDataTypes, IDocumentStore documentStore, out PgQuery pgQuery)
+        {
+            // TODO: Match either \n or \r\n
+            var tableQuery = "select TABLE_SCHEMA, TABLE_NAME, TABLE_TYPE\nfrom INFORMATION_SCHEMA.tables\nwhere TABLE_SCHEMA not in ('information_schema', 'pg_catalog')\norder by TABLE_SCHEMA, TABLE_NAME";
+            if (queryText.Replace("\r", string.Empty) == tableQuery)
+            {
+                pgQuery = new PBIAllCollectionsQuery(queryText, parametersDataTypes, documentStore);
+                return true;
+            }
+
+            pgQuery = null;
+            return false;
+        }
+
+        public override async Task<ICollection<PgColumn>> Init(bool allowMultipleStatements = false)
+        {
+            return new PgColumn[]
```

Comment:
Same comment on compelxity

Reply:
Why is defining the columns in code complex? There aren't many and unfortunately exporting from PgAdmin to CSV only gets you the data and column names, not their types - so this is quite necessary.

Reply:
In your current PR it's OK. See Oren's comment  - "_far_ more readable." https://github.com/omerlagziel/pgrvn/pull/2#discussion_r692103432

Reply:
Done